components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1183,7 +1183,7 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
         }
 
         const sources = await this.workspaceLogService.getWorkspaceLogURLs(wsi);
-        if (!sources) {
+        if (!sources || Object.keys(sources.streams).length === 0) {
```

Comment:
```suggestion
        if (!sources || typeof sources.streams === "object" && Object.keys(sources.streams).length === 0) {
```

just to prevent the `undefined` case